File: 1991-find-the-middle-index-in-array/1991-find-the-middle-index-in-array.py
class Solution:
    def findMiddleIndex(self, nums: List[int]) -> int:
        prefix_sum=0
        total=0
        for i in range(len(nums)):
            total+=nums[i]
        for i in range(len(nums)):
            if(total-prefix_sum-nums[i]==prefix_sum):
                return i
            prefix_sum+=nums[i]
        return -1

File: 1991-find-the-middle-index-in-array/1991-find-the-middle-index-in-array.c
int findMiddleIndex(int* nums, int numsSize) {
        int prefix_sum=0;
        int total=0;
        for(int i=0;i<numsSize;i++)
        {
            total+=nums[i];
        }
        for(int i=0;i<numsSize;i++)
        {
            if(total-prefix_sum-nums[i]==prefix_sum)
            {
                return i;
            }
            prefix_sum+=nums[i];  
        }
        return -1;
}